### ISSUE TITLE
Drinks Recommendation Engine

### ISSUE THREAD

**Issue by gthole** [gthole/drink-stash#6]
- Some kind of distance metric between drinks
- Compare against user ratings
- Suggestions view

**Comment by gthole** [comment]
@jelloeggy I added a new operator to the search syntax as a first step on this: `LIKE list = favorites`, which will find all the recipes that are similar to the ones in a given list (ie have the same or interchangeable ingredients, but possibly different bitters and things) and excludes the recipes that are actually in the list.

On the home page, I changed the "Discover" button to do that search on the user's favorites list instead of the "liquor cabinet" functionality that it had been.  So it gets more interesting as you add recipes to your favorites.

You can also use the search syntax to query for other things "LIKE craigie on main" or "LIKE tiki", which will find all the drinks that are similar to those searches but don't exactly match the search.  (It does slightly strange things when you do "LIKE <ingredient name>", because you're simultaneously filtering and excluding that ingredient.)

Thoughts?

**Comment by jelloeggy** [comment]
Ooo excited to try this! I assume Like works with any list but discover
just uses your Favorites?

**Comment by gthole** [comment]
I've been really enjoying it!

Discover just uses favorites for now.  It's editable in the url, but I plan to add a select box or something to make it easier.  That view really's just a glorified "I'm Feeling Lucky" button.  I think the `LIKE` operator is a lot more useful in the recipes list, and figuring out the search/filter interface to make it more accessible there is something that needs a lot of thought.

**Comment by jelloeggy** [comment]
I am very excited to use the Discover feature. My first foray into it (well, just hitting the button, not trying the drink) has been very exciting. But I'm not sure I understand the LIKE operator. Does it only work with lists? I tried "LIKE prosecco" or "LIKE sparkling" and didn't find anything.

**Comment by gthole** [comment]
Yep it's weird. `LIKE <ingredient>` doesn't make much sense as a search because you're saying "give me all the things that have this ingredient, but then exclude all the things that have that ingredient."  Since `LIKE` excludes everything that was in the original search.

It's primarily meant to work with lists, but can also work with description terms.  I originally thought I'd make it only work for lists, but I got a little carried away with building the search syntax grammar to support combiners and nested queries and stuff.  (Still need to teach it parenthetical precedence, though. :grawesome:)

**Comment by jelloeggy** [comment]
Hmm maybe I need a little cheatsheet on the correct syntax to try out. I just tried LIKE list="Fancy as F*ck" but that didn't work. Ah, without the quotes it worked. LIKE list = Porch Time didn't return anything though. 

I'll keep playing. I love the idea of it.
